src/pipelines/floodscan_pipeline.py

```diff
@@ -0,0 +1,429 @@
+import os
+import re
+import shutil
+from datetime import datetime
+from fileinput import filename
+from zipfile import ZipFile
+
+import pandas as pd
+import requests
+import xarray as xr
+
+from ..utils.azure_utils import blob_client, download_from_azure
+from ..utils.date_utils import (
+    DATE_FORMAT,
+    create_date_range,
+    get_datetime_from_filename,
+)
+from ..utils.raster_utils import invert_lat_lon
+from .pipeline import Pipeline
+
+SFED = "SFED"
+MFED = "MFED"
+
+
+class FloodScanPipeline(Pipeline):
+    def __init__(self, **kwargs):
+        raw_path = kwargs["raw_path"]
+        processed_path = kwargs["processed_path"]
+        kwargs["metadata"]["version"] = kwargs["version"]
+
+        super().__init__(
+            container_name=kwargs["container_name"],
+            raw_path=raw_path,
+            processed_path=processed_path,
+            log_level=kwargs["log_level"],
+            mode=kwargs["mode"],
+            metadata=kwargs["metadata"],
+            use_cache=kwargs["use_cache"],
+        )
+
+        self.start_date = datetime.strptime(kwargs["start_date"], DATE_FORMAT)
+        self.end_date = datetime.strptime(kwargs["end_date"], DATE_FORMAT)
+        self.is_update = kwargs["is_update"]
+        self.is_full_historical_run = kwargs["is_full_historical_run"]
+        self.version = kwargs["version"]
+        self.sfed_historical = kwargs["sfed_historical"]
+        self.mfed_historical = kwargs["mfed_historical"]
+        self.sfed_base_url = os.getenv("FLOODSCAN_SFED_URL")
+        self.mfed_base_url = os.getenv("FLOODSCAN_MFED_URL")
+
+    def _generate_raw_filename(self, date, type):
+        return f"aer_floodscan_{type.lower()}_area_flooded_fraction_africa_90days_{date.strftime(DATE_FORMAT)}.zip"
+
+    def _generate_processed_filename(self, date):
+        return f"aer_area_300s_{date.strftime(DATE_FORMAT)}_v0{self.version}r01.tif"
```

Comment:
Another very minor thing! Could we add a `v` before the date, to follow the convention in [this doc](https://humanitarian.atlassian.net/wiki/spaces/DSCI/pages/3840573448/General+COG+Standards)? So, eg, it would be `aer_area_300s_v2024-01-01_v05r01.tif`?

Reply:
To make the distinction between issued vs valid dates, which is important especially when we have forecasts.

Reply:
Added the v.